src/common/components/elements/ProfileMenu.tsx

```diff
@@ -10,16 +10,32 @@ import { useProfileData } from '@/common/context/ProfileData';
 
 enum TabRoute {
   OWNED,
+  CREATED,
   ACTIVITY,
 }
 
 export const ProfileMenu: FC = () => {
   const { publicKey } = useProfileData();
   const router = useRouter();
   const path = router.asPath;
-  const routeIndex = path.includes('/activity') ? TabRoute.ACTIVITY : TabRoute.OWNED;
+  const pathName = router.pathname;
+  const routeIndex = TabRoute.OWNED;
   const [selectedIndex, setSelectedIndex] = useState(routeIndex);
 
+  useEffect(() => {
+    switch (pathName) {
+      case `/profiles/[publicKey]/created`:
+        setSelectedIndex(TabRoute.CREATED);
+        break;
+      case `/profiles/[publicKey]/activity`:
+        setSelectedIndex(TabRoute.ACTIVITY);
+        break;
+      default:
+        setSelectedIndex(TabRoute.OWNED);
+        break;
+    }
+  }, [pathName, path]);
```

Comment:
Our tabs should be based on the path and when we change tab that should change the route. Doing as an effect with set state leads to unknown conditions on the initial render.

Reply:
@kespinola the  tabs are based on the Routes, this is just to apply "selected" styling to the correct tab